consensus/engine.go

```diff
@@ -0,0 +1,192 @@
+package consensus
+
+import (
+	"sync"
+	"time"
+
+	lru "github.com/hashicorp/golang-lru"
+
+	"github.com/ledgerwatch/turbo-geth/common"
+	"github.com/ledgerwatch/turbo-geth/core/types"
+	"github.com/ledgerwatch/turbo-geth/params"
+)
+
+type VerifyHeaderRequest struct {
+	ID       uint64
+	Headers  []*types.Header
+	Seal     []bool
+	Deadline *time.Time
+}
+
+type VerifyHeaderResponse struct {
+	ID   uint64
+	Hash common.Hash
+	Err  error
+}
+
+type HeadersRequest struct {
+	ID                 uint64
+	HighestHash        common.Hash
+	HighestBlockNumber uint64
+	Number             uint64
+}
+
+type HeaderResponse struct {
+	ID      uint64
+	Headers []*types.Header
+	BlockError
+}
+
+type BlockError struct {
+	Hash   common.Hash
+	Number uint64
+	Err    error
+}
+
+type EngineAPI interface {
+	HeaderVerification() chan<- VerifyHeaderRequest
+	VerifyResults() <-chan VerifyHeaderResponse
+
+	HeaderRequest() <-chan HeadersRequest
+	HeaderResponse() chan<- HeaderResponse
+}
+
+type API struct {
+	Chain                 ChainHeaderReader
+	VerifyHeaderRequests  chan VerifyHeaderRequest
+	VerifyHeaderResponses chan VerifyHeaderResponse
+	CleanupTicker         *time.Ticker
+	CleanupCh             chan FinishedRequest
+	HeadersRequests       chan HeadersRequest
+	HeaderResponses       chan HeaderResponse
+
+	VerifiedBlocks   *lru.Cache // common.Hash->*types.Header
```

Comment:
Looks like the comment is out-of-date - it should be `blockNum -> *types.Header` as per usage below